internal/kgateway/extensions2/plugins/routepolicy/extproc_policy.go

```diff
@@ -87,6 +92,9 @@ func toEnvoyExtProc(
 		envoyExtProc.FailureModeAllow = *extprocConfig.FailureModeAllow
 	}
 
+	if err := envoyExtProc.ValidateAll(); err != nil {
+		return nil, fmt.Errorf("failed to validate envoyExtProc: %s", err.Error())
```

Comment:
nit: Use %v or %w for errors:

```suggestion
		return nil, fmt.Errorf("failed to validate envoyExtProc: %v", err)
```